WooCommerce/Classes/ViewRelated/Orders/OrderListViewModel.swift

```diff
@@ -293,18 +293,21 @@ extension OrderListViewModel {
     ///
     private func bindTopBannerState() {
         let errorState = $hasErrorLoadingData.removeDuplicates()
+        let cppConfiguration = CardPresentConfigurationLoader(stores: stores).configuration
+        let isSupportedCountryForCPP = cppConfiguration.isSupportedCountry
+
         Publishers.CombineLatest3(errorState, $hideOrdersBanners, upsellCardReadersAnnouncementViewModel.$shouldBeVisible)
             .map { hasError, hasDismissedOrdersBanners, upsellCardReadersBannerShouldBeVisible  -> TopBanner in
 
-                guard !hasError else {
+                if hasError {
                     return .error
                 }
 
-                guard !upsellCardReadersBannerShouldBeVisible else {
+                if upsellCardReadersBannerShouldBeVisible && isSupportedCountryForCPP {
```

Comment:
Non a blocker; `upsellCardReadersBannerShouldBeVisible` naming seems a bit confusing now that we added the new condition (the country is supported for CPP), as this is also part of whether the banner should be visible or not. Something like `upsellCardReadersBannerWasDismissed` is more meaningful in my opinion:

`let upsellCardReadersBannerShouldBeVisible = !upsellCardReadersBannerWasDismissed && isSupportedCountryForCPP`

Reply:
I don't think I can turn the meaning around to `wasDismissed` since it comes from the generic `FeatureAnnouncementCardViewModel`. I guess I could do `upsellCardReadersAnnouncementViewModel.$shouldBeVisible.map(!)` but it doesn't seem very readable 😅 

What do you think about `upsellCardReadersBannerAvailable`?

Reply:
> I don't think I can turn the meaning around to wasDismissed since it comes from the generic FeatureAnnouncementCardViewModel. I guess I could do upsellCardReadersAnnouncementViewModel.$shouldBeVisible.map(!) but it doesn't seem very readable 😅

Yeah, that's a good point :) Now that you mention it, perhaps the CPP country availability check should be included in the `FeatureAnnouncementCardViewModel`, or do we want to only check it for orders and not other places such as the Payment Methods Screen? Otherwise `upsellCardReadersBannerAvailable` is a good option 😃

Reply:
Oh I hadn't realized we were showing the upsell in more than one place. In that case, we might want to move the check to `FeatureAnnouncementCardViewModel`. But maybe this is a good start and we can revisit that in another PR 🤷🏽‍♂️

Reply:
Sure 👍